code/dqc/anomalychk.q

```diff
@@ -1,6 +1,14 @@
 \d .dqc
-anomalychk:{[t;colslist;thres]                                                                                  /-function to check percentage of anomalies in each column from colslist of a table t
+anomalychk:{[tn;colslist;thres]
+  / Check percentage of anomalies in each of the columns of t, where the columns 
+  / to watch are specified in colslist, and a percentage threshold, thres.
+  t: get tn;
   d:({sum{any x~'(0w;-0w;0W;-0W)}'[x]}each flip tt)*100%count tt:((),colslist)#t;
   res:([] colsnames:key d; anomalypercentage:value d);
-  update thresholdfail:anomalypercentage>thres from res                                                         /- compare each column's anomalies percentage with threshold thres
+  a:update thresholdfail:anomalypercentage>thres from res;  /- compare each columns anomalies percentage with threshold thres
+  b: exec colsnames from a where anomalypercentage < thresholdfail;
```

Comment:
Don't get this line, `thresholdfail` should be a bool and this checks if `anomalypercentage` is less than 0b or 1b? Combining with `a` would give this oddness:
```suggestion
  b: exec colsnames from a where anomalypercentage < anomalypercentage>thres;
```